pkg/controller/eventhandler.go

```diff
@@ -67,11 +67,61 @@ func (c *Controller) addAllEventHandlers(queueInformer cache.SharedIndexInformer
 			},
 		},
 	)
+
+	queueInformer.AddEventHandler(
+		cache.FilteringResourceEventHandler{
+			FilterFunc: func(obj interface{}) bool {
+				switch obj.(type) {
+				case *v1alpha1.Queue:
+					return true
+				default:
+					return false
+				}
+			},
+			Handler: cache.ResourceEventHandlerFuncs{
+				AddFunc:    c.AddQueue,
+				UpdateFunc: c.UpdateQueue,
+				DeleteFunc: c.DeleteQueue,
+			},
+		},
+	)
+}
+
+func (c *Controller) AddQueue(obj interface{}) {
+	queue := obj.(*v1alpha1.Queue)
+	queueName := queue.Namespace
+	_, ok := c.multiSchedulingQueue.GetQueueByName(queueName)
+	if ok {
+		klog.Errorf("queue is exist %s", queueName)
+		return
+	}
+	err := c.multiSchedulingQueue.Add(queue)
+	if err != nil {
+		klog.Errorf("add queue err %v", err)
+	}
+}
+
+func (c *Controller) UpdateQueue(oldObj, newObj interface{}) {
+	oldQ := oldObj.(*v1alpha1.Queue)
+	newQ := newObj.(*v1alpha1.Queue)
+	err := c.multiSchedulingQueue.Update(oldQ, newQ)
+	if err != nil {
+		klog.Errorf("queue %s update fail %v", oldQ.Namespace, err.Error())
+	}
+}
+
+func (c *Controller) DeleteQueue(obj interface{}) {
+	queue := obj.(*v1alpha1.Queue)
+	err := c.multiSchedulingQueue.Delete(queue)
+	if err != nil {
+		klog.Errorf("queue %s delete fail %v", queue.Namespace, err.Error())
+	}
 }
 
 func (c *Controller) AddQueueUnit(obj interface{}) {
 	unit := obj.(*v1alpha1.QueueUnit)
-	queueName := unit.Spec.Queue
+	// Namespace is key of queueMap
+	queueName := unit.Spec.ConsumerRef.Namespace
```

Comment:
What if not every namespace has its corresponding queue, both of which share a name.

Case 1: admin defines queue `depart-x` for jobs in namespace `sub-group1` and `sub-group2`
Case 2: admin defines queue `vip` for job submitted by certain users.

Reply:
According to our current implementation, cross-namespace queues are not supported now. Because the quota system  we used is resource quota. It is namespace scope. 

But the cross-namespace queues may be support in the future.